src/dioptra/restapi/v1/workflows/service.py

```diff
@@ -115,3 +156,396 @@ def _create_endpoint_analysis_dict(
+
+        Args:
+            group_id: The group to import resources into
+            source_type: The source to import from (either "upload" or "git")
+            git_url: The url to the git repository if source_type is "git"
+            archive_file: The contents of the upload if source_type is "upload_archive"
+            files: The contents of the upload if source_type is "upload_files"
+            config_path: The path to the toml configuration file in the import source.
+            resolve_name_conflicts_strategy: The strategy for resolving name conflicts.
+                Either "fail" or "overwrite"
+
+        Returns:
+            A message summarizing imported resources
+        """
+        log: BoundLogger = kwargs.get("log", LOGGER.new())
+        log.debug("Import resources", group_id=group_id)
+
+        overwrite = (
+            resolve_name_conflicts_strategy
+            == ResourceImportResolveNameConflictsStrategy.OVERWRITE
+        )
+
+        with TemporaryDirectory() as tmp_dir, set_cwd(tmp_dir):
+            working_dir = Path(tmp_dir)
+
+            if source_type == ResourceImportSourceTypes.UPLOAD_ARCHIVE:
+                bytes = archive_file.stream.read()
+                try:
+                    with tarfile.open(fileobj=BytesIO(bytes), mode="r:*") as tar:
+                        tar.extractall(path=working_dir, filter="data")
+                except Exception as e:
+                    raise ImportFailedError("Failed to read uploaded tarfile") from e
+                hash = str(sha256(bytes).hexdigest())
+            elif source_type == ResourceImportSourceTypes.UPLOAD_FILES:
+                hashes = b""
+                for file in files:
+                    Path(file.filename).parent.mkdir(parents=True, exist_ok=True)
+                    bytes = file.stream.read()
+                    with open(working_dir / file.filename, "wb") as f:
+                        f.write(bytes)
+                    hashes = hashes + sha256(bytes).digest()
+                hash = str(sha256(hashes).hexdigest())
+            else:
+                try:
+                    hash = clone_git_repository(cast(str, git_url), working_dir)
+                except Exception as e:
+                    raise GitError(f"Failed to clone repository: {git_url}") from e
+
+            try:
+                with open(working_dir / config_path, "rb") as f:
+                    config = tomllib.load(f)
+            except Exception as e:
+                raise ImportFailedError(
+                    f"Failed to load resource import config from {config_path}."
+                ) from e
+
+            # validate the config file
+            with open(DIOPTRA_RESOURCES_SCHEMA_PATH, "rb") as f:
+                schema = json.load(f)
+            jsonschema.validate(config, schema)
+
+            # all resources are relative to the config file directory
+            os.chdir((working_dir / config_path).parent)
```

Comment:
If we must change the directory, we should control it via a context manager. I don't see this reversed anywhere, so its long-term effect on the REST API's processes/threads could be unpredictable.

The SDK has the following contextmanager, although if we don't want to import from that module within the restapi, we can duplicate it into the restapi utils.py:

https://github.com/usnistgov/dioptra/blob/3b81da427b2a6688a002dc977da2da6bcca70cfd/src/dioptra/sdk/utilities/paths/set_cwd.py#L23-L39